### PR TITLE
bpo-38351: Modernize email example from %-formatting to f-string

### PR DESCRIPTION
**Pull request by dorosch**
https://bugs.python.org/issue38351

### CODE REVIEW

File: Doc/includes/email-dir.py
Comment (brandtbucher) [inline Doc/includes/email-dir.py]:
```suggestion
    msg['Subject'] = f'Contents of directory {context}'
```

File: Doc/includes/email-simple.py
Comment (brandtbucher) [inline Doc/includes/email-simple.py]:
```suggestion
msg['Subject'] = f'The contents of {textfile}'
```

File: Doc/includes/email-unpack.py
Comment (brandtbucher) [inline Doc/includes/email-unpack.py]:
```suggestion
            filename = f'part-{counter:03}{ext}'
```

**Comment by brandtbucher** [comment]
Thanks for the PR @dorosch!

**Review by brandtbucher — CHANGES_REQUESTED** [review]
I agree these should be updated, but [f-strings](https://www.python.org/dev/peps/pep-0498) are actually templated string literals (`str.format` is a third, *different* way of formatting strings).

You're almost there though... see my suggestions below:

**Comment by dorosch** [comment]
> I agree these should be updated, but [f-strings](https://www.python.org/dev/peps/pep-0498) are actually templated string literals (`str.format` is a third, _different_ way of formatting strings).
> 
> You're almost there though... see my suggestions below:

Thanks for your feedback. I used str.format as in other examples of documentation used str.format

File: Doc/includes/email-dir.py
Comment (brandtbucher) [inline Doc/includes/email-dir.py]: Not sure if we need to add this here. It can probably just be inlined as before to keep the diff simple.

```suggestion
```

File: Doc/includes/email-dir.py
Comment (brandtbucher) [inline Doc/includes/email-dir.py]: See above:

```suggestion
    msg['Subject'] = f'Contents of directory {os.path.abspath(directory)}'
```

File: Misc/NEWS.d/next/Documentation/2019-11-15-09-22-28.bpo-38351.xwhlse.rst
Comment (brandtbucher) [inline Misc/NEWS.d/next/Documentation/2019-11-15-09-22-28.bpo-38351.xwhlse.rst]: Minor suggestion here too:

```suggestion
Modernize :mod:`email` examples from %-formatting to f-strings.
```

File: Doc/includes/email-unpack.py
Comment (aeros) [inline Doc/includes/email-unpack.py]: The 'd' should still be included, to explicitly mark it as a decimal. It may not be 100% necessary in this particular case, but I think we should still include it since the original example did. 
```suggestion
            filename = f'part-{counter:03d}{ext}'
```

File: Misc/NEWS.d/next/Documentation/2019-11-15-09-22-28.bpo-38351.xwhlse.rst
Comment (aeros) [inline Misc/NEWS.d/next/Documentation/2019-11-15-09-22-28.bpo-38351.xwhlse.rst]: Edit: I agree with @brandtbucher's suggestion instead.

File: Doc/includes/email-dir.py
Comment (aeros) [inline Doc/includes/email-dir.py]: While I don't have an issue with the changes made (other than the name of the variable perhaps, I'd prefer "path" instead of "context"), it's not related to changing the examples to use f-strings. Also, in this case, I don't think using a variable here makes the example particularly more clear or readable. 

For simplifying the review process, this change to the example should probably be omitted from the PR:
```suggestion
    msg['Subject'] = f'Contents of directory {os.path.abspath(directory)}'
```

(in addition to removing `context = os.path.abspath(directory)` on line 42 of course)

Edit: @brandtbucher's suggestion and mine are identical here, feel free to apply either one.

**Review by brandtbucher — CHANGES_REQUESTED** [review]
Great, just a couple more comments!

**Comment by brandtbucher** [comment]
The CI failure is unrelated. Pushing to this branch should trigger a new build.

**Review by aeros — CHANGES_REQUESTED** [review]
Thanks for the PR @dorosch and welcome! Other than a few minor details, the PR looks most of the way there after @brandtbucher's suggestions.

**Comment by aeros** [comment]
It looks like the travis build failure was not related to the PR, committing the latest suggestions will restart the CI (or closing and reopening the PR).

**Comment by brandtbucher** [comment]
Looks like we crossed posts @aeros. 🙂

**Comment by aeros** [comment]
> Looks like we crossed posts @aeros. slightly_smiling_face

Haha, oops. :slightly_smiling_face:

**Comment by brandtbucher** [comment]
Good to see our comments are almost entirely identical. Haha.

**Comment by dorosch** [comment]
> These changes can be backported to the 3.8 and 3.7 docs as well.

Could you tell me how I can do this?

**Comment by brandtbucher** [comment]
That's a note for the core reviewer. Nothing required on your part @dorosch!

**Review by brandtbucher — APPROVED** [review]
Awesome.

**Review by aeros — APPROVED** [review]
LGTM, thanks for applying the suggested changes @dorosch.

**Comment by aeros** [comment]
@dorosch 
> Could you tell me how I can do this?

@brandtbucher 
> That's a note for the core reviewer. Nothing required on your part @dorosch!

Correct, but more specifically: when the `needs backport to <x>` label is applied, the bot @miss-islington will attempt to automatically open a backport PR to the branch for version <x>, after the original PR is merged. Sometimes when there are merge conflicts, it is required to use the `cherry-picker` tool (https://pypi.org/project/cherry-picker/) to resolve them and manually open the backport PR(s), but most of the time it can be done automatically (especially for minor changes).

**Comment by dorosch** [comment]
> @dorosch
> 
> > Could you tell me how I can do this?
> 
> @brandtbucher
> 
> > That's a note for the core reviewer. Nothing required on your part @dorosch!
> 
> Correct, but more specifically: when the `needs backport to <x>` label is applied, the bot @miss-islington will attempt to automatically open a backport PR to the branch for version `<x>`, after the original PR is merged. Sometimes when there are merge conflicts, it is required to use the `cherry-picker` tool (https://pypi.org/project/cherry-picker/) to resolve them and manually open the backport PR(s), but most of the time it can be done automatically (especially for minor changes).

Thanks for your explanation!

**Review by taleinat — APPROVED** [review]
LGTM

**Review by ericvsmith — APPROVED** [review]

**Comment by brandtbucher** [comment]
Congrats on your first CPython contribution @dorosch! :champagne:

Looking forward to seeing more from you in the future.

**Comment by dorosch** [comment]
> Congrats on your first CPython contribution @dorosch! champagne
> 
> Looking forward to seeing more from you in the future.

@brandtbucher Thank you very much for your help

**Pull request merged**
